### PR TITLE
docs: add desktop session recording and clipboard sharing

### PR DESCRIPTION
**Pull request by zmb3**
Note to reviewers: I figured I would start getting a review on the content before I added screenshots. I'll be sure to add them before this is merged.

Backports required:
- branch/v9

### CODE REVIEW

File: docs/pages/desktop-access/reference/clipboard.mdx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+title: Clipboard Sharing
+description: Using Clipboard Sharing with Teleport Desktop Access.
+---
+
+Teleport Desktop Access supports copying and pasting text between your browser
+and a remote Windows Desktop.
+
+<Adminition type="note">
+This feature requires a Chromium-based browser such as Google Chrome, Brave,
```
Comment (ibeckermayer) [inline docs/pages/desktop-access/reference/clipboard.mdx:10]:
```suggestion
This feature requires a Chromium-based browser such as Brave, Google Chrome,
```

File: docs/pages/desktop-access/reference/clipboard.mdx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+or Microsoft Edge.
+</Admonition>
+
+Clipboard sharing is configured via Teleport's RBAC system. It is enabled by default,
+but can be disabled by setting the following role option on one or more of a user's
+roles:
+
+```yaml
+desktop_clipboard: false
+```
```
Comment (ibeckermayer) [inline docs/pages/desktop-access/reference/clipboard.mdx:20]:
```suggestion
```yaml
kind: role
version: v4
metadata:
  description: Clipboard disabled
  name: clipboard_disabled
spec:
  options:
    desktop_clipboard: false
```
```

File: docs/pages/desktop-access/reference/clipboard.mdx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
```
Comment (ibeckermayer) [inline docs/pages/desktop-access/reference/clipboard.mdx:21]:
```suggestion
<Adminition type="note">
If a user has multiple roles and any one of them sets `desktop_clipboard: false`, clipboard sharing will be disabled for that user.
</Admonition>
```

File: docs/pages/desktop-access/reference/clipboard.mdx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+If a clipboard sharing is enabled for a user, the browser will prompt for access
+to the system clipboard when a desktop session is started. Grant this access to
+proceed.
+
+{
+    /* screenshot of prompt */
+}
+
+- what if unsupported browser
+- what if user clicks no
```
Comment (ibeckermayer) [inline docs/pages/desktop-access/reference/clipboard.mdx:31]:
```suggestion
If you mistakenly deny clipboard permissions when prompted, you can change the setting by clicking the lock on the left side of your address bar, selecting the correct permission, and then refreshing the page
```

Screen Shot 2022-03-10 at 10 08 26

File: docs/pages/desktop-access/reference/sessions.mdx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+## Disable session recording via RBAC
+
+Teleport's RBAC system allows more fine-grained control over which sessions are
+recorded. By default, desktop session recording is enabled. To disable desktop
+session recording for a particular user, add the following role option to all of
+the user's roles:
+
+```yaml
+record_sessions:
+  desktop: false
```
Comment (ibeckermayer) [inline docs/pages/desktop-access/reference/sessions.mdx:39]:
```suggestion
kind: role
version: v4
metadata:
  description: Disable desktop session recording
  name: disable_desktop_recording
spec:
  options:
    record_session:
      desktop: false
```

File: docs/pages/desktop-access/reference/sessions.mdx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+Since Teleport 8.1, sessions can be protected via RBAC. In order for a user to
+see desktop sessions in this list, their roles must permit access to the
+sessions resource.
+
+</Admonition>
+
+Click the play button to open the player in a new tab. The desktop session
+player supports toggling between play and pause, but does not support seeking to
+a specific point in the stream, rewinding, or restarting playback when the end
+of the stream is reached.
```
Comment (ibeckermayer) [inline docs/pages/desktop-access/reference/sessions.mdx:76]:
```suggestion
of the stream is reached. To replay a session, simply refresh the page.
```

File: docs/pages/desktop-access/reference/clipboard.mdx (hunk 1 above)
Comment (ibeckermayer) [reply docs/pages/desktop-access/reference/clipboard.mdx:10]: Haha I know, I'm just trolling

File: docs/pages/desktop-access/reference/clipboard.mdx (hunk 2 above)
Comment (ibeckermayer) [reply docs/pages/desktop-access/reference/clipboard.mdx:20]: That's a valid role spec too, the user can just `tctl create -f` that role and assign it to any user they want to block clipboard sharing from.

File: docs/config.json [hunk 7]
```diff
@@ -517,8 +517,34 @@
...
         },
         {
-          "title": "Configuration",
-          "slug": "/desktop-access/reference/configuration/"
+          "title": "Access Controls",
+          "slug": "/desktop-access/rbac/"
+        },
+        {
+          "title": "Reference",
+          "slug": "/desktop-access/reference/",
```
Comment (ptgott) [inline docs/config.json:525]: This link 404s. I think we should either add a reference.mdx page with a tile for each child page or use the slug for one of the child pages instead.

File: docs/pages/desktop-access/reference/sessions.mdx [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+Recorded sessions can be viewed in the *Session Recordings* page. Desktop
+recordings show a desktop icon in the first column to distinguish them from
+SSH recordings.
+
+{
+    /* screenshot of recordings table */
+}
+
+<Adminition type="note" title="RBAC for sessions">
```
Comment (ptgott) [inline docs/pages/desktop-access/reference/sessions.mdx:66]:
```suggestion
<Admonition type="note" title="RBAC for sessions">
```

File: docs/pages/desktop-access/reference/sessions.mdx [hunk 9]
```diff
@@ -0,0 +1,78 @@
+---
+title: Session Recording and Playback
+description: Recording and Playing Back Teleport Desktop Access Sessions.
+---
+
```
Comment (ptgott) [inline docs/pages/desktop-access/reference/sessions.mdx:5]: Should this be a separate H1 section on disabling session recording?

If so, we'd need to move the first sentence of this next paragraph before the H1 header.

**Comment by zmb3** [comment]
@ibeckermayer @ptgott please take a look at the latest revision - I think I've addressed everything.

**Review by ibeckermayer — APPROVED** [review]

File: docs/pages/desktop-access/reference.mdx [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+    Configure Teleport Desktop Access.
+  </Tile>
+  <Tile icon="list" title="Audit" href="./reference/audit.mdx">
+    Desktop Access audit events.
+  </Tile>
+  <Tile icon="copy" title="Clipboard" href="./reference/clipboard.mdx">
+    Share your clipboard with a remote desktop.
+  </Tile>
+  <Tile icon="play" title="Session Recording" href="./reference/sessions.mdx">
+    Desktop session recording and playback
```
Comment (ptgott) [inline docs/pages/desktop-access/reference.mdx:17]:
```suggestion
    Desktop session recording and playback.
```
for consistency

**Review by ptgott — APPROVED** [review]
Approved with a tiny suggestion

**Pull request merged**